delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,275 @@
+                          case (Some(_), true)    => reject(simultaneousRevAndAncestorsRejection)
+                          case (Some(rev), false) =>
+                            completeIO(
+                              acls
+                                .fetchAt(address, rev)
+                                .map { acl =>
+                                  val searchResults = Seq(acl).flatten
+                                  SearchResults(searchResults.size.toLong, searchResults)
+                                }
+                                .leftWiden[AclRejection]
+                            )
+                          case (None, true)       =>
+                            completeUnscoredSearch(
+                              acls
+                                .fetchWithAncestors(address)
+                                .map(_.value.values.toSeq)
+                            )
+                          case (None, false)      =>
+                            completeUnscoredSearch(
+                              acls
+                                .fetch(address)
+                                .map(Seq(_).flatten)
+                            )
+                        }
+                      }
+                  }
+                )
+              }
+            }
+          },
+          (get & extractAclAddressFilter) { addressFilter =>
+            parameter("self" ? true) {
+              case true  =>
+                completeSearch(
+                  acls
+                    .listSelf(addressFilter)
+                    .map(aclCol => SearchResults(aclCol.value.size.toLong, aclCol.value.values.toSeq))
+                )
+              case false =>
+                completeSearch(
+                  acls
+                    .list(addressFilter)
+                    .map(aclCol => SearchResults(aclCol.value.size.toLong, aclCol.value.values.toSeq))
+                )
+            }
+          }
+        )
+      }
+    }
+  }
+
+}
+
+object AclsRoutes {
+
+  type AclResponseResource = ResourceF[Iri, Acl]
+
+  private val aclsRead  = Permission.unsafe("acls/read")
+  private val aclsWrite = Permission.unsafe("acls/write")
+
+  final private[routes] case class AclEntry(permissions: Set[Permission], identity: Identity)
+  final private[routes] case class AclInput(acl: Seq[AclEntry]) {
+    def toAcl: Acl = Acl(acl.map(entry => entry.identity -> entry.permissions): _*)
+  }
+
+  private[routes] object AclInput {
+    implicit val aclEntryDecoder: Decoder[AclEntry] = deriveDecoder[AclEntry]
+
+    implicit val aclDecoder: Decoder[AclInput] = deriveDecoder[AclInput]
+
+  }
+
+  sealed trait PatchAcl
+  object PatchAcl {
+    final case class SubtractAcl(acl: Acl) extends PatchAcl
+    final case class AppendAcl(acl: Acl)   extends PatchAcl
+
+    implicit val patchAclDecoder: Decoder[PatchAcl] =
+      Decoder.instance { hc =>
```

Comment:
Why don't you use circe configured derivation? you already have encoders and decoders for `AclInput`, and you can have 

```scala
    final case class SubtractAcl(acl: AclInput) extends PatchAcl
    final case class AppendAcl(acl: AclInput)   extends PatchAcl
```

Reply:
I could do something like 
```scala
    final case class Subtract(acl: Seq[AclEntry]) extends PatchAcl
    final case class Append(acl: Seq[AclEntry])   extends PatchAcl
```
I can't use AclInput because the JSON to decode would need too look like:
```json
{
  "acl": {
    "acl": [...]
  }
}
```